heureka/ui/src/components/services/ServicesDetails.jsx

```diff
@@ -3,127 +3,274 @@
+    <Stack direction="vertical" gap="4">
+      <ContentHeading heading="Service Details" />
+
+      <DataGrid columns={2}>
+        <DataGridRow>
+          <DataGridHeadCell className="min-w-[150px]">Owners</DataGridHeadCell>
+          <DataGridCell>
+            {!showComboBox && (
+              <Stack gap="2" wrap={true} alignment="center">
+                {selectedOwners.length > 0 &&
+                  selectedOwners.map((owner, i) => (
+                    <Pill
+                      key={i}
+                      pillValue={owner?.name}
+                      pillValueLabel={owner?.name}
+                      closeable
+                      onClose={() => setConfirmRemoveOwner(owner?.id)}
+                    />
+                  ))}
+                <Icon
+                  color="jn-global-text"
+                  icon="addCircle"
+                  onClick={() => setShowComboBox(true)}
+                  className="ml-2"
+                />
+              </Stack>
+            )}
+
+            {showComboBox && (
+              <Stack direction="horizontal" gap="2">
+                <ComboBox
+                  onChange={(selectedUniqueUserId) => {
+                    const selectedUser = users?.data?.Users?.edges?.find(
+                      (user) => user.node.uniqueUserId === selectedUniqueUserId
+                    )?.node
+                    setNewOwner(selectedUser)
+                  }}
+                >
+                  {users?.data?.Users?.edges
+                    ?.filter(
+                      (user) =>
+                        !selectedOwners.some(
+                          (owner) =>
+                            owner.uniqueUserId === user.node.uniqueUserId
+                        )
                     )
-                  )}
-                </DataGridCell>
-              </DataGridRow>
-            ))}
-          </DataGrid>
-        </>
-      </Stack>
-    </>
+                    .map((user, i) => (
+                      <ComboBoxOption key={i} value={user?.node?.uniqueUserId}>
+                        {user?.node?.name}
+                      </ComboBoxOption>
+                    ))}
+                </ComboBox>
+                <Button variant="primary" onClick={handleAddOwner}>
+                  Save
+                </Button>
+                <Button variant="subdued" onClick={handleCancelAdd}>
+                  Cancel
+                </Button>
+              </Stack>
+            )}
+          </DataGridCell>
+        </DataGridRow>
+
+        {/* Modal for remove confirmation */}
+        {confirmRemoveOwner && (
+          <>
+            <Modal
+              cancelButtonLabel="Cancel"
+              confirmButtonLabel="Yes, Proceed"
+              open={!!confirmRemoveOwner}
+              onCancel={() => setConfirmRemoveOwner(null)}
+              onConfirm={() => handleRemoveOwner(confirmRemoveOwner)}
+            >
+              <p>Are you sure you want to remove this owner?</p>
```

Comment:
It's always better to repeat to the user what exactly is getting removed. So pass the user name to the modal and then change the text to `Are you sure you want to remove { ownerName } from owners?`